### DISCUSSION TITLE
Recommended way for installing php-cs-fixer

### DISCUSSION THREAD

**Discussion by arturkhamaiun** [PHP-CS-Fixer/PHP-CS-Fixer#6866] [Q&A] [answered]
Hi,

I want to install php-cs-fixer and the documentation suggests me to install it to a separate folder in a project:
```
mkdir -p tools/php-cs-fixer
composer require --working-dir=tools/php-cs-fixer friendsofphp/php-cs-fixer
```

but I don't understand the reason for such a decision. Could you please explain why installing into a separate folder and not using the main composer.json file is recommended?  Like below:
```
composer require friendsofphp/php-cs-fixer --dev
```

**Comment by Wirone** [comment, marked as answer]
Installing Fixer in a separate folder makes your projects' actual dependencies less prone to conflicts with Fixer's dependencies.

You should _not_ be blocked from updating some of your required packages just because your dev tool does not support higher versions of that packages. That's why installing dev tool in separate directory is handy - you have encapsulated dependencies for the tool, while your main projects has smaller set of dependencies to resolve.

**Reply by keradus** [reply]
You can also use shim repo, which is having all dependencies built into phar file:
https://packagist.org/packages/php-cs-fixer/shim
